### PR TITLE
feat(utilities): change push/pad/pull utility baselines to be based on 4px

### LINKED ISSUE #919: Push/pad/pull utility baselines should be based on 4px instead of 5px
#### Bug Report

Currently the [utility styles](https://teradata.github.io/covalent/#/style-guide/utility-styles), such as `.pad-sm`, `.push-lg`, and `.pull-right` are implemented based on a 5px baseline.

#### What is the expected behavior?
Utility styles should be implemented in multiples of 4px.

#### What is the motivation / use case for changing the behavior?
The [material design spec](https://material.io/guidelines/layout/metrics-keylines.html#metrics-keylines-baseline-grids) indicates that

> All components align to an 8dp square baseline grid for mobile, tablet, and desktop. Iconography in toolbars align to a 4dp square baseline grid.

The 5px baseline makes the utility styles inaccurate for adjustments that conform to the material spec.

### PR DESCRIPTION
**Pull request by aciccarello**
## Description

Aligns utility baselines to the [material design spec](https://material.io/guidelines/layout/metrics-keylines.html#metrics-keylines-baseline-grids), which indicates

> All components align to an 8dp square baseline grid for mobile, tablet, and desktop. Iconography in toolbars align to a 4dp square baseline grid.

This PR also tries to clean up the utility styles and improve the documentation.

Closes #919

### Breaking Changes
The `.push-bottom-button`, `.pad-bottom-button`, and `.pull-bottom-xxxl` classes have been removed. Additionally, the sizes of the `.push`, `.pad`, and `.pull` classes has changed slightly.

- push/pad/pull none `0 -> 0` (no change)
- push/pad/pull xs `5px -> 4px`
- push/pad/pull sm `10px -> 8px`
- push/pad/pull (no size) `20px -> 16px`
- push/pad md `30px -> 24px`
- push/pad lg `40px -> 32px`
- push/pad xl `50px -> 48px`
- push/pad xxl `60px -> 56px`
- pull md `-15px -> -24px`
- pull lg `-20px -> -32px`
- pull xl `-30px -> -48px`
- pull xxl `-40px -> -56px`

### What's included?

- [x] Remove unused `*-bottom-button` and `*-xxxl` styles
- [x] Change push/pad/pull utility sizes
- [x] Align `.pull-*` style sizes to `.push` and `.pad`
- [x] Add units to style-guide/utility-styles page
- [x] Increase messages component padding
- [x] Adjust docs padding/margin
- [x] Use SCSS to generate classes at compile time _(optional)_

#### Test Steps

- [x] start dev server
- [x] check units listed on utility styles docs
- [x] check messages component
- [x] ensure margin/padding of docs are properly aligned

#### General Tests for Every PR

- [x] `ng serve --aot` still works.
- [x] `npm run lint` passes.
- [x] `npm test` passes and code coverage is not lower.
- [x] `npm run build` still works.

##### Screenshots or link to CodePen/Plunker/JSfiddle
Updated documentation
image

Existing messages component vs updated:
image

### CODE REVIEW

**Comment by aciccarello** [comment]
The `.pull` sizes are different than `.push`. Oddly the `.push-md` is larger than the `.push` class but the `.pull-md` is smaller than the `.pull` class. Additionally `.pull-lg` is the same as `.pull`.

I would suggest aligning the sizes exactly though the larger `.pull` sizes wouldn't see much use.

Other planned work is listed in `What's Included?` section. I'll build and review the docs for alignment issues once I know what to do with the `.push` styles.

**Comment by emoralesb05** [comment]
Agree, they should be consistent

**Comment by emoralesb05** [comment]
Is this ready for review @aciccarello ?

**Comment by aciccarello** [comment]
Sorry, not yet. I'm planning on updating the pull sizes which will require fixing some of the docs. I'll send out a message on slack when it's ready. Let me know if there's a time crunch.

**Comment by aciccarello** [comment]
I've aligned the pull sizes to push and pad. I still need to review the docs to see if the changes are causing any significant margin issues.

I'm open to feedback on the updated docs and message component. I'm wondering if the docs should have a separate card with the sizes listed once rather than repeated for each css class. For the message component I increased the margin because having only 8px of padding on the top and bottom seemed small. Therefore I tried to align the sizes to the spec of the toast message which oddly uses a 14px margin. Another thing to consider is removing the utility styles from any of the actual components. It seems that only the message and dynamic slider use them.

File: src/platform/core/message/message.component.html [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <div tdMessageContainer></div>
 <ng-template>
   <div layout="column">
-    <div class="pad-top-sm pad-right pad-bottom-sm pad-left td-message-wrapper" layout="row" layout-align="center center">
-      <mat-icon class="push-right">{{icon}}</mat-icon>
+    <div class="pad pad-right-md pad-left-md td-message-wrapper" layout="row" layout-align="center center">
```
Comment (emoralesb05) [inline src/platform/core/message/message.component.html]: I think we can leave this the way it was, the reduced px's make it look a bit better in my opinion

**Pull request merged**